### ISSUE TITLE
Network requests not blocked for short period after browser start. 

### ISSUE THREAD

**Issue by kasper93** [gorhill/uBlock#2970]
### Describe the issue

Looks like browser can bypass uBlock for the short period of time, probably before extension is fully initialized. I noticed this when restarting browser for update, on session restore page will have ads not blocked.

### One or more specific URLs where the issue occurs

I tested with `hltv.org` but any site will do.

### Steps for anyone to reproduce the issue

1. Go to any page where network request is blocked by uBlock.
1. Restart browser - SHIFT+F2 and type `restart --nocache`
1. Browser will restart and restore previous tabs. But loaded tab will not have ads blocked, because network request were made probably before uBlock is ready. 

### Your settings

- OS/version: Windows 10
- Browser/version: Firefox Nightly (20170904100131)
- uBlock Origin version:  1.14.8

**Comment by Atavic** [comment]
It's a browser issue, open it in offline mode to prevent possible leaks.

**Comment by Atavic** [comment]
Also see: https://github.com/gorhill/uBlock/issues/1913

**Comment by uBlock-user** [comment]
Open `Advanced settings` page and set `suspendTabsUntilReady` to `true` and restart browser.

**Comment by gorhill** [comment]
Duplicate of #1327.

**Comment by gwarser** [comment]
>Open `Advanced settings` page, set `suspendTabsUntilReady` to `true` and restart browser.

Sadly, not work form me any more.

- Cookies must be allowed on UBO settings page, for this settings to be remembered between sessions.
- Even with cookies allowed I end up with first tab and address bar being blank. URL only flashes for a moment and then disappear leaving first visible tab unusable.

**Comment by gorhill** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1378459

**Comment by gorhill** [comment]
Something else I could experiment with: Firefox's webRequest API allows to return Promise to tell whether a request should be blocked or not. uBO could make use of this and resolve all the pending Promises once all the filters are loaded. I have no clue if that would work though, just that conceptually that seems right.

**Comment by luxoflux** [comment]
@gorhill
Did you ever get around to trying that experiment?

**Comment by luxoflux** [comment]
I found that setting the browser to clear cache on exit fixes this issue for me. Good enough.

Tested with 57b9 and 58.0a1 (2017-10-19).
